ui/webui/src/apis/boss.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with This program; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+
+/**
+ * @param {string} address      Anaconda bus address
+ *
+ * @returns {Object}            A DBus client for the Boss bus
+ */
+export const bossClient = ({ address }) => cockpit.dbus(
+    "org.fedoraproject.Anaconda.Boss",
+    { superuser: "try", bus: "none", address }
```

Comment:
Just wondering, while this is likely not related to the JS code, but the DBus API itself, what does the superuser: try mean & why do we put "none" as the first tuple member we feed into `address` ? Of is it "none" fed to `bus` and address is a positional argument somehow ?

Reply:
For the cockpit.dbus arguments documentation read: https://cockpit-project.org/guide/latest/cockpit-dbus.html

in the section -> cockpit.dbus()

The superuser will just try to create the bus with root privileged. I guess it's actually no needed, as these processes are running in privileged mode.

"none" is for the "bus" configuration. we need to put "none" according to the docs above, so as to connect to the non-standard bus specified with the address option. This defaults to "system".

The address value in the object will be substituted by the 'address' parameter of the bossClient function.

Reply:
I get it now, thanks! :)